From 91561b269db5531882ad8054b6b1ad58afb7f72d Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 26 Sep 2020 10:17:41 +0200
Subject: [PATCH 1/1] riscv: reset after crash

If an exception occurs on ARM or x86, we call panic() which will try to
reset the board. Do the same on RISC-V.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 arch/riscv/lib/interrupts.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/riscv/lib/interrupts.c b/arch/riscv/lib/interrupts.c
index 8ff40f0f36..f95cb97f92 100644
--- a/arch/riscv/lib/interrupts.c
+++ b/arch/riscv/lib/interrupts.c
@@ -84,7 +84,7 @@ static void _exit_trap(ulong code, ulong epc, ulong tval, struct pt_regs *regs)
 
 	show_regs(regs);
 	show_efi_loaded_images(epc);
-	hang();
+	panic("Resetting CPU ...\n");
 }
 
 int interrupt_init(void)
-- 
2.28.0

